### ISSUE TITLE
Create a unit UI scene

### ISSUE THREAD

**Issue by WildWeazel** [C7-Game/Prototype#170]
Create a single reusable and scriptable Godot scene to display a more faithful unit UI: a bordered column of HP pegs and status indicator, both color-coded, and containers for the unit and active unit cursor animations. The scene should have functions to set the current and max HP, border, status, color scheme, and play animations, both programmatically and via editor controls.

**Comment by lukesgray** [comment]
I can take this issue on.

**Comment by WildWeazel** [comment]
@lukesgray just checking in, any progress or questions?

**Comment by lukesgray** [comment]
@WildWeazel I have still be doing some tutorials in Godot. I'm trying to lean on the built-in features and libraries as much as possible as that seems like best practice to me but that means it will take me a little longer to youtube what I need to since I don't have prior Godot experience.

**Comment by WildWeazel** [comment]
No problem, and if you'd like to start even smaller I'm sure we can find more UI improvements.

**Comment by lukesgray** [comment]
That may be a good idea. At this point I'm not going to be able to be a reliable contributor. Maybe in a few months but for now, I'd say feel free to take me off of this issue or leave me on until someone else wants it or I have time to tackle it. Or yeah, something else smaller would be fine too!

**Comment by WildWeazel** [comment]
Of course. @pcen does anything come to mind? I'm sure there are some UI items on the main screen that could use some work.

**Comment by pcen** [comment]
I think we still need a standardised button scene / component. I'm not really sure when to use a scene vs. just write a class that inherits from a Godot class. Currently our buttons are just modified `TextureButton` instances but we should probably have a button class/scene because a lot of buttons in civ3 look the same.

When implementing this, something to consider is how we're loading textures in general, if a mod changes the button texture, what will the codepath be that determines which texture to give a component (ie. buttons).

**Comment by WildWeazel** [comment]
#177 covers reusable buttons. #175 may also be a candidate.
